torch_xla/csrc/reduction.cpp

```diff
@@ -31,7 +31,9 @@ ReductionInfo GetReductionInfo(
       rinfo.new_dimensions.push_back(shape.dimensions(i));
     }
   }
-  XLA_CHECK_GT(rinfo.element_count, 0);
+  if (check_positive_numel) {
```

Comment:
This should not be here IMHO.
I mean, we end up adding one line anyway at call site, so be better do the check over there.

Reply:
Getting element_count in each call site wasn't free :P Or am I missing an API in `xla::Shape` that can do it?

Reply:
It's part of `rinfo`, isn't it?